src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/ForceMergeActionIT.kt

```diff
@@ -0,0 +1,222 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.action
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.IndexStateManagementRestTestCase
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.makeRequest
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.Policy
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.State
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.Transition
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.DeleteActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ForceMergeActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ReadOnlyActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.randomDefaultNotification
+import org.elasticsearch.rest.RestStatus
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.Locale
+
+class ForceMergeActionIT : IndexStateManagementRestTestCase() {
+
+    private val testIndexName = javaClass.simpleName.toLowerCase(Locale.ROOT)
+
+    fun `test basic workflow`() {
+        val indexName = "${testIndexName}_index"
+        val policyID = "${testIndexName}_testPolicyName"
+
+        // Create a Policy with two States, one that force_merges and one that deletes the index
+        val forceMergeActionConfig = ForceMergeActionConfig(maxNumSegments = 1, index = 0)
+        val deleteActionConfig = DeleteActionConfig(0)
+        val states = listOf(
+            State(
+                "ForceMergeState",
+                listOf(forceMergeActionConfig),
+                listOf(Transition(stateName = "DeleteState", conditions = null))
+            ),
+            State("DeleteState", listOf(deleteActionConfig), listOf())
+        )
+
+        val policy = Policy(
+            id = policyID,
+            description = "$testIndexName description",
+            schemaVersion = 1L,
+            lastUpdatedTime = Instant.now().truncatedTo(ChronoUnit.MILLIS),
+            defaultNotification = randomDefaultNotification(),
+            defaultState = states[0].name,
+            states = states
+        )
+
+        createPolicy(policy, policyID)
+        createIndex(indexName, null)
+        refresh()
+
+        // Add sample data to increase segment count, passing in a delay so that multiple segments get created
+        insertSampleData(indexName, 3, 1000)
+
+        val segmentCount = getSegmentCount(indexName)
+        assertTrue("Segment count for [$indexName] was less than expected", segmentCount > 1)
+
+        // Add policy to index after data is loaded
```

Comment:
You should be able to just add the policyID on the initial createIndex. It takes 1 minute to initialize policy and 1 minute to execute the first action so you shouldn't have any issues with policy being run too soon.